### PR TITLE
Improve BWD / FWD_BWD accuracy checks; fix layer_norm bwd check

### PR DESCRIPTION
**Pull request by yf225**
Test: `python run.py --op layer_norm --bwd --num-inputs 1 --metrics accuracy`

### CODE REVIEW

File: tritonbench/operators/layer_norm/operator.py [hunk 1]
```diff
@@ -82,9 +82,37 @@ def liger_layer_norm(self, *args):
         return lambda: LigerLayerNormFunction.apply(x, weight, bias, eps)
 
     def get_bwd_fn(self, fwd_fn: Callable) -> Callable:
+        from torch.utils._pytree import tree_map
+
+        # Run forward once to get output
         y = fwd_fn()
         dy = 0.1 * torch.randn_like(y)
-        return lambda: y.backward(dy, retain_graph=True)
```
Comment (yf225) [inline tritonbench/operators/layer_norm/operator.py:87]: I believe usually to check backward accuracy, we need to check the gradients of the forward input tensors. But the previous design doesn't allow us to do it.
